### PR TITLE
CA-442-When-Conf-App-ended-and-user-logged-in-bring-the-user-to-TABula-main-screen1

### PR DESCRIPTION
**Pull request by giladadon1997**
**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [ ] I acknowledge that all my contributions will be made under the project's license.

### JIRA link(s):

- [AHOYAPPS-0000](https://issues.corp.twilio.com/browse/AHOYAPPS-0000)

### Description

A description of what this PR does.

### Before review
* [ ] Updated CHANGELOG.md if necessary
* [ ] Added unit tests if necessary
* [ ] Updated affected documentation
* [ ] Verified locally with `npm test`
* [ ] Manually sanity tested running locally
* [ ] Included screenshot as PR comment (if needed)
* [ ] Ready for review

### Before merge
* [ ] Got one or more +1s
* [ ] Re-tested if necessary

### CODE REVIEW

**Review by talr — CHANGES_REQUESTED** [review]
Why it is to merge-code branch and not to integration (develop) branch?
Tal

File: src/state/index.tsx [hunk 1]
```diff
@@ -95,17 +89,16 @@ export default function AppStateProvider(props: React.PropsWithChildren<{}>) {
...
         console.log('response body from fetch: ' + JSON.stringify(responseBodyAsJson));
-        //setEndpoint(responseBodyAsJson.endPoint);
         endpoint = responseBodyAsJson.endPoint;
+        environmentName = responseBodyAsJson.environmentName;
       });
   }
 
   async function ensureEndpointInitialised() {
-    if (endpoint === '') {
+    if (endpoint === '' || environmentName === '') {
```
Comment (talr) [inline src/state/index.tsx]: Can we replace all the console.log() which eScribers Logger?

File: src/state/index.tsx [hunk 2]
```diff
@@ -174,6 +167,14 @@ export default function AppStateProvider(props: React.PropsWithChildren<{}>) {
 
       return data;
     },
+    disconnectParticipant: async (isRegistered?: boolean) => {
+      if (!(await ensureEndpointInitialised())) return null;
+      var decodedRedirectTabulaUrl = atob(returnUrl ? returnUrl : '');
+      var loginPageUrl = `http://tabula-${environmentName}.escribers.io/tabula/welcome/login`;
+      console.log('loginPageUrl:' + loginPageUrl);
+      if (isRegistered) window.location.replace(decodedRedirectTabulaUrl);
```
Comment (talr) [inline src/state/index.tsx:185]: Can we ident the code? i.e. this if statement

File: src/state/index.tsx [hunk 3]
```diff
@@ -174,6 +167,14 @@ export default function AppStateProvider(props: React.PropsWithChildren<{}>) {
 
       return data;
     },
+    disconnectParticipant: async (isRegistered?: boolean) => {
+      if (!(await ensureEndpointInitialised())) return null;
+      var decodedRedirectTabulaUrl = atob(returnUrl ? returnUrl : '');
+      var loginPageUrl = `http://tabula-${environmentName}.escribers.io/tabula/welcome/login`;
```
Comment (talr) [inline src/state/index.tsx]: In production it is not escribers.io but escribers.net, please talk with Chana to get this from the deployment pipeline.

**Review by talr — CHANGES_REQUESTED** [review]

**Comment by giladadon1997** [comment]
> Why it is to merge-code branch and not to integration (develop) branch?
> Tal

merge-code branch is the develop, for now, we can change it, but we need to ask Chana before, because its depend in cicd pipeline

File: src/state/index.tsx (hunk 1 above)
Comment (giladadon1997) [reply src/state/index.tsx]: yes, it is another story in next sprint

File: src/state/index.tsx (hunk 3 above)
Comment (giladadon1997) [reply src/state/index.tsx]: I talked about it with Chana, at the moment the environments are not aligned so she told me to leave like this, later we will change.

File: src/state/index.tsx (hunk 2 above)
Comment (giladadon1997) [reply src/state/index.tsx:185]: what line do you talk about?

File: src/state/index.tsx (hunk 3 above)
Comment (talr) [reply src/state/index.tsx]: @chani3251 Why not to do it correctly?

**Pull request merged**